kingfisher_scrapy/spiders/malta.py

```diff
@@ -28,9 +35,15 @@ def start_requests(self):
     @handle_http_error
     def parse_list(self, response):
         urls = response.json()['packagesPerMonth']
-
         netloc = urlsplit(response.request.url).netloc
         for url in reversed(urls):
-            # URL looks like http://malta-demo-server.eurodyn.com/ocds/services/recordpackage/getrecordpackage/2020/1
+            if self.from_date and self.until_date:
+                # URL looks like http://malta-demo-server.eurodyn.com/ocds/services/recordpackage/getrecordpackage/
+                # 2020/1
+                year = int(url.split('/')[-2])
+                month = int(url.split('/')[-1])
```

Comment:
Split once (also for Zambia)

```suggestion
                year, month = map(int, url.rsplit('/', 2)[1:])
```